### PR TITLE
For/stable/joshua s brown fix abs

### PR DESCRIPTION
**Pull request by JoshuaSBrown**
The wrong abs function was being used. Somewhere a c header was being included that was converting a double to int when calling abs. This was mitigated by using fabs.

The bug was discovered as a result in the lammps data reader in csg: https://github.com/votca/csg/issues/537

### CODE REVIEW

File: include/votca/tools/akimaspline.h [hunk 1]
```diff
@@ -98,8 +98,8 @@ inline double AkimaSpline::getSlope(double m1, double m2, double m3,
   if ((m1 == m2) && (m3 == m4)) {
```
Comment (junghans) [inline include/votca/tools/akimaspline.h:98]: We should fix this comparison, too.

Comment (JoshuaSBrown) [reply include/votca/tools/akimaspline.h:98]: Wow you are fast. I just created this.

Comment (JoshuaSBrown) [reply include/votca/tools/akimaspline.h:98]: And what would be a suitable tolerance? I have no idea what sizes these numbers typically have.

**Comment by JensWehner** [comment]
Heyho I am not convinced. I do not see how that should happen, std::abs is templated.

**Comment by junghans** [comment]
I think replacing `abs()` by `std::abs()` should be enough.

**Comment by JoshuaSBrown** [comment]
Ok, std::abs then, we should just be careful when using abs though. @JensWehner which part are you not convinced about?

**Comment by junghans** [comment]
And can we add a test to csg, comparing the output of csg_dump.

**Comment by JensWehner** [comment]
works for me

**Comment by JensWehner** [comment]
we should try to remove all using namespace std;

**Comment by JoshuaSBrown** [comment]
> And can we add a test to csg, comparing the output of csg_dump.

I don't think this is necessary, this errors is actually caught in the tools unit element test as it is.

File: CMakeLists.txt [hunk 2]
```diff
@@ -54,7 +54,7 @@ endif(IS_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/.git)
 
 find_package(Threads REQUIRED)
 
-find_package(Boost 1.53.0 REQUIRED COMPONENTS program_options filesystem system )
+find_package(Boost 1.63.0 REQUIRED COMPONENTS program_options filesystem system )
```
Comment (junghans) [inline CMakeLists.txt:57]: I think that might be a bad idea, we still need to support bionic for @baumeier, which only has boost-1.65: https://packages.ubuntu.com/bionic/libboost-all-dev

Comment (JoshuaSBrown) [reply CMakeLists.txt:57]: Ok

File: src/libtools/histogram.cc [hunk 3]
```diff
@@ -76,7 +76,7 @@ void Histogram::ProcessData(DataCollection<double>::selection* data) {
   if (_options._scale == "bond") {
     for (size_t i = 0; i < _pdf.size(); ++i) {
       double r = _min + _interval * (double)i;
-      if (abs(r) < 1e-10) {
+      if (fabs(r) < 1e-10) {
```
Comment (junghans) [inline src/libtools/histogram.cc:79]: `std::` here?

File: src/libtools/histogram.cc [hunk 4]
```diff
@@ -86,7 +86,7 @@ void Histogram::ProcessData(DataCollection<double>::selection* data) {
     for (size_t i = 0; i < _pdf.size(); ++i) {
       double alpha = _min + _interval * (double)i;
       double sa = sin(alpha);
-      if (abs(sa) < 1e-5) {
+      if (fabs(sa) < 1e-5) {
```
Comment (junghans) [inline src/libtools/histogram.cc:89]: `std::` here?

**Comment by JensWehner** [comment]
I think we should not allows using namspace std;. CLang-tidy flags it as well.

**Comment by junghans** [comment]
> I think we should not allows using namspace std;. CLang-tidy flags it as well.

This is for stable, let's fix it in master.

File: src/libtools/histogram.cc (hunk 4 above)
Comment (JoshuaSBrown) [reply src/libtools/histogram.cc]: done

File: src/libtools/histogram.cc (hunk 3 above)
Comment (JoshuaSBrown) [reply src/libtools/histogram.cc]: done

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,10 @@ For more detailed information about the changes see the history of the [reposito
 * fix build with mkl (#229)
 * fix build with non-system libfftw (#234)
 * fix CI on Ubuntu-20.04 (#237)
+* fix bug related to calling c abs instead of c++ fabs (#248)
+* updated floating point comparison to use boost epsilon error (#248)
+* promoted boost version from 1.53 to 1.63 (#248)
```
Comment (junghans) [inline CHANGELOG.md:9]: This can go!

File: include/votca/tools/structureparameters.h [hunk 6]
```diff
@@ -105,7 +105,7 @@ class StructureParameters {
   T get(const StructureParameter parameter) const;
 
  private:
-  std::unordered_map<StructureParameter, boost::any> parameters;
+  std::unordered_map<StructureParameter, boost::any, std::hash<int>> parameters;
```
Comment (junghans) [inline include/votca/tools/structureparameters.h:108]: Why is that needed?

File: CHANGELOG.md (hunk 5 above)
Comment (JoshuaSBrown) [reply CHANGELOG.md:9]: Shoot.

File: include/votca/tools/structureparameters.h (hunk 6 above)
Comment (JoshuaSBrown) [reply include/votca/tools/structureparameters.h:108]: It turns out tools was also broken on ubuntu 16 in an unrelated issue. This fixes that.

Comment (junghans) [reply include/votca/tools/structureparameters.h:108]: Ok, Ubuntu 16 is pretty old, correct? Was the boost bump related?

Comment (JoshuaSBrown) [reply include/votca/tools/structureparameters.h:108]: Um no, that was for the floating point comparison issue in akimaspline. I wanted to use the boost epsilon which is only available starting in boost 1.63

Comment (JoshuaSBrown) [reply include/votca/tools/structureparameters.h:108]: Yes, 16 has been out for a while.
